### PR TITLE
feat: delete stale users job

### PR DESCRIPTION
**Pull request by Mautjee**
## Description

This PR introduces a cronjob in the Postgres Database. This cron job runs every 24 hours on 0:10 GMT and deletes the user which do not have a username and Aleo address. 
 
## Related Issues

#156 

## Changes Made

- [BE] Created a new docker file for the postgres instance

### CODE REVIEW

File: postgres.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM ankane/pgvector
+
+RUN apt-get update && apt-get install -y --no-install-recommends postgresql-15-cron
+
+COPY ./db-init.sh /docker-entrypoint-initdb.d/db-init.sh
```
Comment (ramidecodes) [inline postgres.Dockerfile:5]: What about executing the .sh commands in here, instead of having a diff file?

Comment (Mautjee) [reply postgres.Dockerfile:5]: We need to run these commands after Postgres is initialised and not at build time.

> The PostgreSQL data directory (/var/lib/postgresql/data) is usually not created until the PostgreSQL server is initialized, which typically happens when the container is run, not when it's being built.

The files in the `docker-entrypoint-initb.d/` folder  will run directly when Postgres is initialised

Comment (Mautjee) [reply postgres.Dockerfile:5]: I can look into a work around so we have every thing in one file

Comment (Mautjee) [reply postgres.Dockerfile:5]: Found a solution

**Review by ramidecodes — APPROVED** [review]

File: Dockerfile.postgres [hunk 2]
```diff
@@ -0,0 +1,7 @@
+FROM ankane/pgvector
+
+RUN apt-get update && apt-get install -y --no-install-recommends postgresql-15-cron
+
+RUN echo "shared_preload_libraries = 'pg_cron'" >> /usr/share/postgresql/postgresql.conf.sample
+
+COPY ./pg-init.sql /docker-entrypoint-initdb.d/pg_init.sql
```
Comment (silimarius) [inline Dockerfile.postgres:7]: Finish with empty line

File: src/server/db/user.ts [hunk 3]
```diff
@@ -30,7 +30,7 @@ export const selectMatchPlayedByUser = async (userId: string) => {
     .select()
     .from(matches)
     .where(
-      sql`id IN (SELECT unnest(matches_played) AS id FROM bot_busters_user WHERE id = ${userId})`
+      sql`${matches.id} IN (SELECT unnest(${users.matchesPlayed}) AS id FROM ${users} WHERE id = ${userId})`
```
Comment (silimarius) [inline src/server/db/user.ts:33]: Do this in `storeScore` function in `src/server/service/match.ts` as well

File: pg-init.sql [hunk 4]
```diff
@@ -0,0 +1,4 @@
+CREATE EXTENSION IF NOT EXISTS pg_cron;
+CREATE EXTENSION IF NOT EXISTS vector;
+
+SELECT cron.schedule('0 10 * * *', $$DELETE FROM bot_busters_user WHERE bot_busters_user.created_at < NOW() - INTERVAL '24 HOURS' AND bot_busters_user.address IS NULL AND bot_busters_user.username IS NULL$$);
```
Comment (silimarius) [inline pg-init.sql:4]: Finish with empty line

File: Dockerfile.postgres [hunk 5]
```diff
@@ -0,0 +1,7 @@
+FROM ankane/pgvector
```
Comment (silimarius) [inline Dockerfile.postgres:1]: Specify version

**Review by silimarius — APPROVED** [review]

**Pull request merged**